src/background/stores/path-snippet-store.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { StoreNames } from '../../common/stores/store-names';
+import { PathSnippetData } from '../../common/types/store-data/path-snippet-data';
+import { PathSnippetActions } from '../actions/path-snippet-actions';
+import { BaseStoreImpl } from './base-store-impl';
+
+export class PathSnippetStore extends BaseStoreImpl<PathSnippetData> {
+    private pathSnippetActions: PathSnippetActions;
+
+    constructor(pathSnippetActions: PathSnippetActions) {
+        super(StoreNames.PathSnippetStore);
+
+        this.pathSnippetActions = pathSnippetActions;
+    }
+
+    public getDefaultState(): PathSnippetData {
+        const defaultValue: PathSnippetData = {
+            path: '',
+            snippet: '',
+        };
+
+        return defaultValue;
+    }
+
+    protected addActionListeners(): void {
+        this.pathSnippetActions.onAddPath.addListener(this.onAddPath);
+        this.pathSnippetActions.onAddSnippet.addListener(this.onAddSnippet);
+    }
+
+    private onAddPath = (payload: string): void => {
+        this.state.path = payload;
+        this.emitChanged();
+    };
+
+    private onAddSnippet = (payload: string): void => {
```

Comment:
Having this 2 functions (`onAddPath` and `onAddSnippet`) is fine here but consider to have a more generic version of them. Something around this:
```typescript
    protected addActionListeners(): void {
        this.pathSnippetActions.onAddPath.addListener(payload => this.onChangeProperty('path', payload));
        this.pathSnippetActions.onAddSnippet.addListener(payload => this.onChangeProperty('snippet', payload));
    }

    private onChangeProperty = (property: keyof PathSnippetData, payload: string) => {
        this.state[property] = payload;
    }
```

This is only possible because there is not complex, different logic between updating the path and the snippet. It's less code and, if for any reason, we need to add a new string property (it should even work with other types) this will minimize future changes to the class

Reply:
Ah got it. Thanks!